website/server/libs/api-v3/cron.js

```diff
@@ -110,6 +110,7 @@ export async function recoverCron (status, locals) {
 export function cron (options = {}) {
   let {user, tasksByType, analytics, now = new Date(), daysMissed, timezoneOffsetFromUserPrefs} = options;
 
+  user.lastCron = now; // TODO does putting this here help prevent this double-cron bug: https://github.com/HabitRPG/habitrpg/issues/2805#issuecomment-222314260
```

Comment:
this is not setting lastCron at the beginning of the process but only when the user is saved